packages/enzyme/src/ShallowWrapper.js

```diff
@@ -243,6 +246,52 @@ function privateSetChildContext(adapter, wrapper, instance, renderedNode, getChi
   }
 }
 
+/**
+ * Recursively dive()s every custom component in a wrapper until
+ * the target component is found.
+ *
+ * @param {ShallowWrapper} wrapper A ShallowWrapper to search
+ * @param {ComponentType} target A react custom component that, when found, will end recursion
+ * @param {Adapter} adapter An Enzyme adapter
+ * @returns {ShallowWrapper|undefined} A ShallowWrapper for the target, or
+ *  undefined if it can't be found
+ */
+function deepRender(wrapper, target, adapter) {
+  const node = wrapper[NODE];
+  const element = node && adapter.nodeToElement(node);
+  if (wrapper.type() === target) {
+    return wrapper.dive();
+  }
+  if (element && isCustomComponentElement(element, adapter)) {
+    return deepRender(wrapper.dive(), target, adapter);
+  }
+  const children = wrapper.children();
+  for (let i = 0; i < children.length; i += 1) {
+    const found = deepRender(children.at(i), target, adapter);
+    if (typeof found !== 'undefined') {
+      return found;
+    }
+  }
+  return undefined;
+}
+
+/**
+ * Deep-renders the `wrappingComponent` and returns the context that should
+ * be accessible to the primary wrapper.
+ *
+ * @param {WrappingComponentWrapper} wrapper The `WrappingComponentWrapper` for a
+ *  `wrappingComponent`
+ * @param {Adapter} adapter An Enzyme adapter
+ * @returns {object} The context collected
+ */
+function getContextFromWrappingComponent(wrapper, adapter) {
+  const rootFinder = deepRender(wrapper, wrapper[ROOT_FINDER], adapter);
+  if (!rootFinder) {
+    throw new Error('`wrappingComponent` must render its children!');
+  }
+  return rootFinder[OPTIONS].context;
+}
```

Comment:
Same with this function. It operates entirely with `ShallowWrapper` APIs—nothing react-specific.